pydis_site/apps/content/templatetags/page_github.py

```diff
@@ -0,0 +1,21 @@
+import requests
+from django import template
+
+register = template.Library()
+
+
+@register.filter()
+def page_github(current_uri: str) -> str:
+    """
+    Return the corresponding GitHub page for the current site page.
+
+    Return a directory if the page is a directory on GitHub otherwise return the Markdown file
+    """
+    github_uri = current_uri.replace(
+        "http://pythondiscord.local:8000/pages/",
+        "https://github.com/python-discord/site/tree/main/pydis_site/apps/content/resources/"
+    )
+    github_uri = github_uri[:-1] if github_uri.endswith("/") else github_uri
+    if requests.get(github_uri).status_code == 404:
+        github_uri += ".md"
```

Comment:
I think this can be refactored to avoid an unnecessary HTTP request to Github.

We can already differentiate between pages and categories in the view (`pydis_site/apps/content/views/page_category.py`), so we could for example add the template name to the context in `get_context_data`, then use that to determine the URI path for the Github link.